numba/tests/test_unicode.py

```diff
@@ -337,6 +349,55 @@ def test_find(self, flags=no_pyobj_flags):
                                  cfunc(a, substr),
                                  "'%s'.find('%s')?" % (a, substr))
 
+    def test_rfind(self):
+        pyfunc = rfind_usecase
+        cfunc = njit(pyfunc)
+
+        subs = [(s, ['', 'xx', s[:-2], s[3:], s]) for s in UNICODE_EXAMPLES]
+        # Samples taken from CPython testing:
+        # https://github.com/python/cpython/blob/201c8f79450628241574fba940e08107178dc3a5/Lib/test/test_unicode.py#L233-L259
+        cpython_subs = [
+            ('\u0102' + 'a' * 100, ['\u0102', '\u0201', '\u0120', '\u0220']),
+            ('\U00100304' + 'a' * 100, ['\U00100304', '\U00100204', '\U00102004']),
+            ('abcdefghiabc', ['abc', '']),
+            ('a' + '\u0102' * 100, ['a']),
+            ('a' + '\U00100304' * 100, ['a']),
+            ('\u0102' + '\U00100304' * 100, ['\u0102']),
+            ('a' * 100, ['\u0102', '\U00100304', '\u0102a', '\U00100304a']),
+            ('\u0102' * 100, ['\U00100304', '\U00100304\u0102']),
+            ('_a' + '\u0102' * 100, ['_a']),
+            ('_a' + '\U00100304' * 100, ['_a']),
+            ('_\u0102' + '\U00100304' * 100, ['_\u0102']),
+        ]
+        for s, subs in subs + cpython_subs:
+            for sub_str in subs:
+                msg = 'Results of interpreted and compiled "{}".rfind("{}") should be equal'.format(s, sub_str)
+                self.assertEqual(pyfunc(s, sub_str), cfunc(s, sub_str), msg = msg)
+
+    def test_rfind_with_start_end(self):
+        pyfunc = rfind_with_start_end_usecase
+        cfunc = njit(pyfunc)
+
+        for s in UNICODE_EXAMPLES:
+            subs = ['', 'xx', s[:-2], s[3:], s]
+            for sub_str in subs:
```

Comment:
Probably, use  ['', 'xx', s[:-2], s[3:], s] in loop definition?

Reply:
Ok.